gridstatus/caiso.py

```diff
@@ -1460,19 +1460,19 @@ def _make_timestamp(time_str, today, timezone="US/Pacific"):
     return ts
 
 
-def _get_historical(file, date, verbose=False):
-    try:
-        date_str = date.strftime("%Y%m%d")
-        url = _HISTORY_BASE + "/%s/%s.csv" % (date_str, file)
-        msg = f"Fetching URL: {url}"
-        log(msg, verbose)
-    except Exception:
-        # fallback if today and no historical file yet
-        if utils.is_today(date, CAISO.default_timezone):
-            url = _BASE + "/%s.csv" % file
-            msg = f"Fetching URL: {url}"
-            log(msg, verbose)
-
+def _get_historical(
+    file: str,
+    date: str | pd.Timestamp,
+    verbose: bool = False,
+) -> pd.DataFrame:
+
+    if utils.is_today(date, CAISO.default_timezone):
```

Comment:
What happens if the file for today is not available yet?

Reply:
Conceivably there is always a file available at the https://www.caiso.com/outlook/sp/demand.csv URL, which we assume is for "today" (not sure how valid that assumption is, tho)

Reply:
Ahh, okay, I see, thanks. It's the historical file for a given date that may not be available yet.